paltas/Sampling/distributions.py

```diff
@@ -408,3 +495,111 @@ def __call__(self):
 			list(float): |num| samples from dist
 		"""
 		return self.dist(size=self.num)
+
+
+# PSF Generator
+
+def xy_helper(x,y):
+	# 1st 7 are bottom row, 2nd 7 are 2nd row, etc...
+	return int(((y%4)*7)-1 + x%7)
+
+
+class PSFGenerator():
+	"""
+	Args:
+		psf_filepath: .fits file that stores PSFs
+		coords_dict dict{scipy.stats.rv_continuous.rvs}: dict of 
+			distributions to sample for coordinates 
+		psf_option (string): option for how PSF file is handled
+
+    Notes:
+        psf_option 'emp_f814w' requires coord_dict keys: 'CCD', 'x', 'y', 'focus'
+	"""
+
+	def __init__(self,psf_filepath,coords_dict,psf_option='emp_f814w'):
+		self.psf_filepath = psf_filepath
+		with fits.open(self.psf_filepath) as hdu:
+			self.psf_kernels = hdu[0].data
+		self.coords_dict = coords_dict
+		self.psf_option = psf_option
+
+	def hst_emp_f814w_mapping(self,coords):
```

Comment:
This code looks a little scary: lots of math, hardcoded numbers, print statements, etc. It also seems pretty hubble specific. If there is no way to use an external library for some of this (e.g. maybe numpy or scipy for the interpolation?), you could move the bulk of this function to a separate function in `hubble_utils.py` that you can call here. And Sebastian would want you to add unit tests for this :-)